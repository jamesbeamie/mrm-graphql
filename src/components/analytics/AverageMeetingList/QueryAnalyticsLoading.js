import React from 'react';
import ProgressBar from 'react-toolbox/lib/progress_bar';
import TableHead from '../../helpers/TableHead';
import '../../../assets/styles/averageMeetinglist.scss';

/**
 * Component for loading query analytics
 *
 * @returns {JSX}
 */
const QueryAnalyticsLoading = () => (
  <div className="average-meeting">
    <div className="average-meeting-control">
      <h4 className="header-title">Average time spent/Meeting Room</h4>
    </div>
    <div className="average-meeting-list">
      <div className="table">
        <TableHead
          titles={['Room', 'No. of meetings', 'Average Meeting Duration']}
        />
        <div className="table__body">
          <div className="table__row--loading">
            <ProgressBar type="linear" mode="indeterminate" />
          </div>
        </div>
      </div>
      <br />
    </div>
  </div>
);

export default QueryAnalyticsLoading;
