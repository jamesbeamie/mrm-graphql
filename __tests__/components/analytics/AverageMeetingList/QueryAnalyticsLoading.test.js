import React from 'react';
import { shallow } from 'enzyme';
import QueryAnalyticsLoading from '../../../../src/components/analytics/AverageMeetingList/QueryAnalyticsLoading';

describe('QueryAnalyticsLoading component', () => {
  const wrapper = shallow(<QueryAnalyticsLoading />);
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
