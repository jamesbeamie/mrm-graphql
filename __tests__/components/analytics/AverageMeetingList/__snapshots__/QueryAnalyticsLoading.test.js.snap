// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryAnalyticsLoading component renders correctly 1`] = `
<div
  className="average-meeting"
>
  <div
    className="average-meeting-control"
  >
    <h4
      className="header-title"
    >
      Average time spent/Meeting Room
    </h4>
  </div>
  <div
    className="average-meeting-list"
  >
    <div
      className="table"
    >
      <TableHead
        titles={
          Array [
            "Room",
            "No. of meetings",
            "Average Meeting Duration",
          ]
        }
      />
      <div
        className="table__body"
      >
        <div
          className="table__row--loading"
        >
          <ThemedProgressBar
            buffer={0}
            className=""
            composeTheme="deeply"
            mapThemrProps={[Function]}
            max={100}
            min={0}
            mode="indeterminate"
            multicolor={false}
            type="linear"
            value={0}
          />
        </div>
      </div>
    </div>
    <br />
  </div>
</div>
`;
